src/Unsafe/Coerce.purs

```diff
@@ -8,20 +8,84 @@ module Unsafe.Coerce
 -- | (that is, the caller's) responsibility to ensure that the underlying
 -- | representation for both types is the same.
 -- |
--- | One application for this function is to avoid doing work that you know is a
--- | no-op because of newtypes. For example, if you have an `Array (Conj a)` and you
--- | want an `Array (Disj a)`, you could do `Data.Array.map (runConj >>> Disj)`, but
--- | this performs an unnecessary traversal. `unsafeCoerce` accomplishes the same
--- | for free.
--- |
--- | It is highly recommended to define specializations of this function rather than
--- | using it as-is. For example:
--- |
--- | ```purescript
--- | mapConjToDisj :: forall a. Array (Conj a) -> Array (Disj a)
--- | mapConjToDisj = unsafeCoerce
+-- | Note: because this function is extraordinarily flexible, type inference
+-- | can greatly suffer. Whenever you use it, you should add type annotations
+-- | to clarify what the type of it's input value is and the type of it's
+-- | output value.
+-- |
+-- | After the v0.14.0 PureScript release, some of what was accomplished via
+-- | `unsafeCoerce` can now be accomplished via `coerce` from
+-- | `purescript-safe-coerce`. See that library's documentation for more
+-- | context.
+-- |
+-- | There are at least two ways you might want to use this function:
+-- | 1. To model existentials
+-- | 2. To model the kinds of arguments an FFI function can take if one or
+-- |    more of its arguments can be multiple types.
+-- |
+-- | ## Modeling Existentials
```

Comment:
I think `unsafeCoerce` is really just an implementation detail of these two ideas; it's not fundamentally tied to either of them. For instance, existentials can also be implemented without any escape hatches like unsafeCoerce, and at some point in the future, we might implement them in the language, which would mean that there would be no need to use this technique at all. Therefore, I'm not sure this repo's API reference is a good place to discuss them in such depth. I think material discussing these techniques would be better placed either in the guides directory of the documentation repo, or in a cookbook recipe.

Reply:
Good point. I'll remove these two examples, but keep that note about the type inference.